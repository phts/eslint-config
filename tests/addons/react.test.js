'use strict'

const helpers = require('../helpers')

describe('addons/react', () => {
  helpers.itGeneratesCorrectConfig({
    extends: ['index', 'addons/react'],
    files: ['src/target.js', 'src/target.jsx', 'src/dir/target.js', 'src/dir/target.jsx'],
    for: 'src/**/*.{js,jsx}',
  })
})
