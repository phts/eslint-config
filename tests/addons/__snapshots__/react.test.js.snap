// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addons/react generates correct config for src/**/*.{js,jsx} files 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
  },
  "globals": Object {},
  "parser": "node_modules/babel-eslint/lib/index.js",
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": true,
    },
    "ecmaVersion": 2018,
    "sourceType": "module",
  },
  "plugins": Array [
    "eslint-comments",
    "import",
    "react",
  ],
  "rules": Object {
    "accessor-pairs": Array [
      "error",
    ],
    "array-bracket-spacing": Array [
      "error",
    ],
    "brace-style": Array [
      "error",
    ],
    "camelcase": Array [
      "error",
    ],
    "comma-dangle": Array [
      "error",
      "always-multiline",
    ],
    "comma-spacing": Array [
      "error",
    ],
    "comma-style": Array [
      "error",
    ],
    "constructor-super": Array [
      "error",
    ],
    "curly": Array [
      "error",
    ],
    "dot-location": Array [
      "error",
      "property",
    ],
    "eol-last": Array [
      "error",
    ],
    "eqeqeq": Array [
      "error",
    ],
    "eslint-comments/disable-enable-pair": Array [
      "error",
      Object {
        "allowWholeFile": true,
      },
    ],
    "eslint-comments/no-aggregating-enable": Array [
      "error",
    ],
    "eslint-comments/no-duplicate-disable": Array [
      "error",
    ],
    "eslint-comments/no-unlimited-disable": Array [
      "error",
    ],
    "eslint-comments/no-unused-disable": Array [
      "error",
    ],
    "eslint-comments/no-unused-enable": Array [
      "error",
    ],
    "for-direction": Array [
      "error",
    ],
    "func-call-spacing": Array [
      "error",
    ],
    "getter-return": Array [
      "error",
    ],
    "import/export": Array [
      "error",
    ],
    "import/first": Array [
      "error",
    ],
    "import/named": Array [
      "error",
    ],
    "import/newline-after-import": Array [
      "error",
    ],
    "import/no-absolute-path": Array [
      "error",
    ],
    "import/no-cycle": Array [
      "error",
    ],
    "import/no-named-as-default": Array [
      "error",
    ],
    "import/no-named-as-default-member": Array [
      "error",
    ],
    "import/no-self-import": Array [
      "error",
    ],
    "import/no-useless-path-segments": Array [
      "error",
    ],
    "import/order": Array [
      "error",
    ],
    "indent": Array [
      "error",
      2,
    ],
    "jsx-quotes": Array [
      "error",
      "prefer-double",
    ],
    "key-spacing": Array [
      "error",
      Object {
        "afterColon": true,
        "beforeColon": false,
        "mode": "minimum",
      },
    ],
    "keyword-spacing": Array [
      "error",
    ],
    "max-len": Array [
      "error",
      Object {
        "code": 100,
      },
    ],
    "new-cap": Array [
      "error",
    ],
    "new-parens": Array [
      "error",
    ],
    "no-array-constructor": Array [
      "error",
    ],
    "no-async-promise-executor": Array [
      "error",
    ],
    "no-caller": Array [
      "error",
    ],
    "no-case-declarations": Array [
      "error",
    ],
    "no-class-assign": Array [
      "error",
    ],
    "no-compare-neg-zero": Array [
      "error",
    ],
    "no-cond-assign": Array [
      "error",
    ],
    "no-console": Array [
      "error",
      Object {
        "allow": Array [
          "info",
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": Array [
      "error",
    ],
    "no-constant-condition": Array [
      "error",
    ],
    "no-control-regex": Array [
      "error",
    ],
    "no-debugger": Array [
      "error",
    ],
    "no-delete-var": Array [
      "error",
    ],
    "no-dupe-args": Array [
      "error",
    ],
    "no-dupe-class-members": Array [
      "error",
    ],
    "no-dupe-keys": Array [
      "error",
    ],
    "no-duplicate-case": Array [
      "error",
    ],
    "no-duplicate-imports": Array [
      "error",
    ],
    "no-empty": Array [
      "error",
    ],
    "no-empty-character-class": Array [
      "error",
    ],
    "no-empty-pattern": Array [
      "error",
    ],
    "no-eval": Array [
      "error",
    ],
    "no-ex-assign": Array [
      "error",
    ],
    "no-extra-bind": Array [
      "error",
    ],
    "no-extra-boolean-cast": Array [
      "error",
    ],
    "no-extra-parens": Array [
      "error",
      "all",
      Object {
        "ignoreJSX": "all",
      },
    ],
    "no-extra-semi": Array [
      "error",
    ],
    "no-fallthrough": Array [
      "error",
    ],
    "no-floating-decimal": Array [
      "error",
    ],
    "no-func-assign": Array [
      "error",
    ],
    "no-global-assign": Array [
      "error",
    ],
    "no-implied-eval": Array [
      "error",
    ],
    "no-inner-declarations": Array [
      "error",
    ],
    "no-invalid-regexp": Array [
      "error",
    ],
    "no-irregular-whitespace": Array [
      "error",
    ],
    "no-label-var": Array [
      "error",
    ],
    "no-labels": Array [
      "error",
    ],
    "no-lone-blocks": Array [
      "error",
    ],
    "no-misleading-character-class": Array [
      "error",
    ],
    "no-mixed-spaces-and-tabs": Array [
      "error",
    ],
    "no-multiple-empty-lines": Array [
      "error",
    ],
    "no-new-object": Array [
      "error",
    ],
    "no-new-require": Array [
      "error",
    ],
    "no-new-symbol": Array [
      "error",
    ],
    "no-obj-calls": Array [
      "error",
    ],
    "no-octal": Array [
      "error",
    ],
    "no-path-concat": Array [
      "error",
    ],
    "no-proto": Array [
      "error",
    ],
    "no-prototype-builtins": Array [
      "error",
    ],
    "no-redeclare": Array [
      "error",
    ],
    "no-regex-spaces": Array [
      "error",
    ],
    "no-return-assign": Array [
      "error",
    ],
    "no-return-await": Array [
      "error",
    ],
    "no-self-assign": Array [
      "error",
    ],
    "no-self-compare": Array [
      "error",
    ],
    "no-sequences": Array [
      "error",
    ],
    "no-shadow-restricted-names": Array [
      "error",
    ],
    "no-sparse-arrays": Array [
      "error",
    ],
    "no-tabs": Array [
      "error",
    ],
    "no-this-before-super": Array [
      "error",
    ],
    "no-throw-literal": Array [
      "error",
    ],
    "no-trailing-spaces": Array [
      "error",
    ],
    "no-undef": Array [
      "error",
    ],
    "no-undef-init": Array [
      "error",
    ],
    "no-unexpected-multiline": Array [
      "error",
    ],
    "no-unmodified-loop-condition": Array [
      "error",
    ],
    "no-unneeded-ternary": Array [
      "error",
    ],
    "no-unreachable": Array [
      "error",
    ],
    "no-unsafe-finally": Array [
      "error",
    ],
    "no-unsafe-negation": Array [
      "error",
    ],
    "no-unused-labels": Array [
      "error",
    ],
    "no-unused-vars": Array [
      "error",
    ],
    "no-useless-call": Array [
      "error",
    ],
    "no-useless-catch": Array [
      "error",
    ],
    "no-useless-computed-key": Array [
      "error",
    ],
    "no-useless-constructor": Array [
      "error",
    ],
    "no-useless-escape": Array [
      "error",
    ],
    "no-useless-rename": Array [
      "error",
    ],
    "no-var": Array [
      "error",
    ],
    "no-whitespace-before-property": Array [
      "error",
    ],
    "no-with": Array [
      "error",
    ],
    "object-curly-spacing": Array [
      "error",
      "never",
    ],
    "object-property-newline": Array [
      "error",
      Object {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false,
      },
    ],
    "one-var": Array [
      "error",
      "never",
    ],
    "one-var-declaration-per-line": Array [
      "error",
    ],
    "padded-blocks": Array [
      "error",
      "never",
    ],
    "prefer-const": Array [
      "error",
    ],
    "quote-props": Array [
      "error",
      "as-needed",
    ],
    "quotes": Array [
      "error",
      "single",
      Object {
        "avoidEscape": true,
      },
    ],
    "react/boolean-prop-naming": Array [
      "off",
    ],
    "react/button-has-type": Array [
      "off",
    ],
    "react/default-props-match-prop-types": Array [
      "off",
    ],
    "react/destructuring-assignment": Array [
      "off",
    ],
    "react/display-name": Array [
      "off",
    ],
    "react/forbid-component-props": Array [
      "off",
    ],
    "react/forbid-dom-props": Array [
      "off",
    ],
    "react/forbid-elements": Array [
      "off",
    ],
    "react/forbid-foreign-prop-types": Array [
      "off",
    ],
    "react/forbid-prop-types": Array [
      "off",
    ],
    "react/jsx-boolean-value": Array [
      "off",
    ],
    "react/jsx-child-element-spacing": Array [
      "off",
    ],
    "react/jsx-closing-bracket-location": Array [
      "off",
    ],
    "react/jsx-closing-tag-location": Array [
      "off",
    ],
    "react/jsx-curly-brace-presence": Array [
      "off",
    ],
    "react/jsx-curly-spacing": Array [
      "off",
    ],
    "react/jsx-equals-spacing": Array [
      "off",
    ],
    "react/jsx-filename-extension": Array [
      "off",
    ],
    "react/jsx-first-prop-new-line": Array [
      "off",
    ],
    "react/jsx-fragments": Array [
      "off",
    ],
    "react/jsx-handler-names": Array [
      "off",
    ],
    "react/jsx-indent": Array [
      "off",
    ],
    "react/jsx-indent-props": Array [
      "off",
    ],
    "react/jsx-key": Array [
      "error",
    ],
    "react/jsx-max-depth": Array [
      "off",
    ],
    "react/jsx-max-props-per-line": Array [
      "off",
    ],
    "react/jsx-no-bind": Array [
      "off",
    ],
    "react/jsx-no-comment-textnodes": Array [
      "error",
    ],
    "react/jsx-no-duplicate-props": Array [
      "error",
    ],
    "react/jsx-no-literals": Array [
      "off",
    ],
    "react/jsx-no-target-blank": Array [
      "error",
    ],
    "react/jsx-no-undef": Array [
      "error",
    ],
    "react/jsx-one-expression-per-line": Array [
      "off",
    ],
    "react/jsx-pascal-case": Array [
      "off",
    ],
    "react/jsx-props-no-multi-spaces": Array [
      "off",
    ],
    "react/jsx-sort-default-props": Array [
      "off",
    ],
    "react/jsx-sort-props": Array [
      "off",
    ],
    "react/jsx-space-before-closing": Array [
      "off",
    ],
    "react/jsx-tag-spacing": Array [
      "off",
    ],
    "react/jsx-uses-react": Array [
      "error",
    ],
    "react/jsx-uses-vars": Array [
      "error",
    ],
    "react/jsx-wrap-multilines": Array [
      "off",
    ],
    "react/no-access-state-in-setstate": Array [
      "off",
    ],
    "react/no-array-index-key": Array [
      "off",
    ],
    "react/no-children-prop": Array [
      "error",
    ],
    "react/no-danger": Array [
      "off",
    ],
    "react/no-danger-with-children": Array [
      "error",
    ],
    "react/no-deprecated": Array [
      "error",
    ],
    "react/no-did-mount-set-state": Array [
      "off",
    ],
    "react/no-did-update-set-state": Array [
      "off",
    ],
    "react/no-direct-mutation-state": Array [
      "error",
    ],
    "react/no-find-dom-node": Array [
      "error",
    ],
    "react/no-is-mounted": Array [
      "error",
    ],
    "react/no-multi-comp": Array [
      "off",
    ],
    "react/no-redundant-should-component-update": Array [
      "off",
    ],
    "react/no-render-return-value": Array [
      "error",
    ],
    "react/no-set-state": Array [
      "off",
    ],
    "react/no-string-refs": Array [
      "error",
    ],
    "react/no-this-in-sfc": Array [
      "off",
    ],
    "react/no-typos": Array [
      "off",
    ],
    "react/no-unescaped-entities": Array [
      "error",
    ],
    "react/no-unknown-property": Array [
      "error",
    ],
    "react/no-unsafe": Array [
      "off",
    ],
    "react/no-unused-prop-types": Array [
      "off",
    ],
    "react/no-unused-state": Array [
      "off",
    ],
    "react/no-will-update-set-state": Array [
      "off",
    ],
    "react/prefer-es6-class": Array [
      "off",
    ],
    "react/prefer-read-only-props": Array [
      "off",
    ],
    "react/prefer-stateless-function": Array [
      "off",
    ],
    "react/prop-types": Array [
      "off",
    ],
    "react/react-in-jsx-scope": Array [
      "error",
    ],
    "react/require-default-props": Array [
      "off",
    ],
    "react/require-optimization": Array [
      "off",
    ],
    "react/require-render-return": Array [
      "error",
    ],
    "react/self-closing-comp": Array [
      "off",
    ],
    "react/sort-comp": Array [
      "off",
    ],
    "react/sort-prop-types": Array [
      "off",
    ],
    "react/state-in-constructor": Array [
      "off",
    ],
    "react/style-prop-object": Array [
      "off",
    ],
    "react/void-dom-elements-no-children": Array [
      "off",
    ],
    "require-atomic-updates": Array [
      "error",
    ],
    "require-yield": Array [
      "error",
    ],
    "rest-spread-spacing": Array [
      "error",
    ],
    "semi": Array [
      "error",
      "never",
    ],
    "semi-spacing": Array [
      "error",
    ],
    "space-before-blocks": Array [
      "error",
    ],
    "space-before-function-paren": Array [
      "error",
      Object {
        "anonymous": "always",
        "asyncArrow": "always",
        "named": "never",
      },
    ],
    "space-in-parens": Array [
      "error",
    ],
    "space-infix-ops": Array [
      "error",
    ],
    "space-unary-ops": Array [
      "error",
    ],
    "spaced-comment": Array [
      "error",
    ],
    "strict": Array [
      "error",
    ],
    "template-curly-spacing": Array [
      "error",
    ],
    "use-isnan": Array [
      "error",
    ],
    "valid-typeof": Array [
      "error",
    ],
    "wrap-iife": Array [
      "error",
    ],
    "yield-star-spacing": Array [
      "error",
      "before",
    ],
    "yoda": Array [
      "error",
    ],
  },
  "settings": Object {
    "react": Object {
      "version": "detect",
    },
  },
}
`;
